src/main/java/org/zalando/nakadi/service/EventTypeService.java

```diff
@@ -126,10 +129,17 @@ public EventTypeService(final EventTypeRepository eventTypeRepository,
         this.etLogEventType = etLogEventType;
         this.eventTypeOptionsValidator = eventTypeOptionsValidator;
         this.adminService = adminService;
+        this.authorizationService = authorizationService;
     }
 
     public List<EventType> list() {
-        return eventTypeRepository.list();
+        return authorizationService.filter(eventTypeRepository.list()
```

Comment:
@lmontrieux I think that for event types this new filter interface adapted very well. But I'm thinking about subscriptions, which are paginated and more numerous. For example, if a user asks for a page with 10 results, then we fetch from the DB 10 resources, filter out, we are left with 2. Then we have to fetch the next bunch, until we potentially had to load the entire DB.

I thought about this problem for a while and I cannot think of an easy solution for that. How to apply the filter to the query itself? Only if we used some kind of ORM and we passed the query to the plugin and let it manipulate it 😨 

That said, I think we can live with this inefficiency.